### DISCUSSION TITLE
OpenAI API Key is deleted each time the Docker Container is restarted

### DISCUSSION THREAD

**Discussion by KaiserRuben** [open-webui/open-webui#2347] [General]
# Bug Report

## Description

**Bug Summary:**
Given you entered an OpenAI API Key, once you restart the Docker Container, its gone and you'd need to configure it again.

**Steps to Reproduce:**
1. Start Open-WebUI using the Docker Container
2. Add an OpenAI API Key in the settings under "Connections"
3. Restart the Docker Container

**Expected Behavior:**
I would expect the API Key to be persistent, so still be there even after the Docker container is restarted.

**Actual Behavior:**
Well, it is not persistent. Its gone when you restart the Container (or your PC for that matter)

## Environment

- **Open WebUI Version:** v0.1.124
- **Ollama (if applicable):** 0.1.37 (but doesn't matter for this)

- **Operating System:** MacOS 14.5 (23F79)
- **Browser (if applicable):** Arc Version 1.43.0 (50018); (Chromium Engine Version 125.0.6422.60)

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.

## Installation Method

Using [Open-WebUIs official website](https://docs.openwebui.com/getting-started/): 
```sh
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by justinh-rahb** [comment]
https://docs.openwebui.com/getting-started/env-configuration#openai

**Reply by KaiserRuben** [reply]
Thanks!

But does this solve the problem though? I still believe it would be more intuitive to have it being persistent.

If you give me a starting point where I'd need to look, I am happy to think about a fix. 👍🏼😊

**Reply by justinh-rahb** [reply]
Coming in a future update, it's already in dev branch.
